src/InlineAutocomplete/types.ts

```diff
@@ -0,0 +1,56 @@
+import {ActionListItemProps} from '../ActionList'
+
+export interface Trigger {
+  /** A single character that can cause the suggestion list to open. */
+  triggerChar: string
+  /**
+   * Control whether the suggestion query can contain spaces. This should *not* be `true` if
+   * `triggerChar` is a space.
+   * @default false
+   */
+  multiWord?: boolean
+  /**
+   * Control whether the trigger character is retained when inserting a suggestion.
+   * @default true
+   */
+  keepTriggerCharOnCommit?: boolean
+}
+
+export interface ShowSuggestionsEvent {
+  /** The trigger that caused this query. */
+  trigger: Trigger
+  /** The query string. */
+  query: string
+}
+
+export type Suggestion =
+  | string
+  | {
+      /**
+       * The plain text value of the suggestion. This is the text that will be inserted when
+       * the user applies the suggestion. If no `key` is provided, this value **must** be unique
+       * across all currently visible suggestions.
+       */
+      value: string
+      /**
+       * Optional key. If not provided, the `value` will be used. Setting a `key` allows
+       * for non-unique `value`s.
+       */
+      key?: string
+      /** This must return an `ActionList.Item` instance. */
+      render: (props: ActionListItemProps) => React.ReactElement
```

Comment:
I don't love `render` functions and I know that's not really the Primer way, but I was unable to think of a better API. We could theoretically use composition here, but the list is pretty much always going to be an array that is then mapped to components - I feel like that would generally be less readable. I'm definitely open to suggestions.

Reply:
Resolving - discussed elsewhere